test/Microsoft.AspNet.WebUtilities.Tests/QueryHelpersTests.cs

```diff
@@ -52,5 +52,39 @@ public void ParseQueryWithEmptyKeyWorks()
             Assert.Equal(1, collection.Count);
             Assert.Equal(new[] { "value1", "" }, collection[""]);
         }
+
+        [Theory]
+        [InlineData("http://contoso.com/", "http://contoso.com/?hello=world")]
+        [InlineData("http://contoso.com/someaction", "http://contoso.com/someaction?hello=world")]
+        [InlineData("http://contoso.com/someaction?q=test", "http://contoso.com/someaction?q=test&hello=world")]
+        [InlineData(
+            "http://contoso.com/someaction?q=test#anchor",
+            "http://contoso.com/someaction?q=test&hello=world#anchor")]
+        [InlineData("http://contoso.com/someaction#anchor", "http://contoso.com/someaction?hello=world#anchor")]
+        [InlineData("http://contoso.com/#anchor", "http://contoso.com/?hello=world#anchor")]
+        public void AddQueryStringWithKeyAndValue(string uri, string expectedUri)
```

Comment:
No, `http://contoso.com/someaction?q=test#anchor?value` + `name=value` => `http://contoso.com/someaction?q=test&name=value#anchor?value`. Only the first `?` and `#` in a url are special.
Also consider:
`http://contoso.com/someaction#anchor?stuff` + `name=value` => `http://contoso.com/someaction?name=value#anchor?stuff`